pkg/util/priority_queue.go

```diff
@@ -42,9 +45,10 @@ func (q *queue) Pop() interface{} {
 }
 
 // NewPriorityQueue makes a new priority queue.
-func NewPriorityQueue() *PriorityQueue {
+func NewPriorityQueue(length prometheus.Gauge) *PriorityQueue {
```

Comment:
I think it would be nice if this was optional since it is a util. Also, I would prefer a different name such as `lengthGauge` so it is not confused with a cached integer length for some reason.